com.examples.exercise/src/test/java/com/examples/exercise/testStudentController.java

```diff
@@ -10,33 +10,33 @@
 public class testStudentController {
 	private StudentController studControl;
 	private StudentsDB myStudentsDB;
-	
+
 	@Before
-	public void setup(){
+	public void setup() {
 		myStudentsDB = mock(StudentsDB.class);
-		studControl=new StudentController(myStudentsDB);
+		studControl = new StudentController(myStudentsDB);
 	}
 
 	@Test
 	public void testUpdateIterationWithDB() {
 		studControl.update("00000", "matteo");
 		verify(myStudentsDB, times(1)).exists(anyString());
-		
+
 	}
-	
+
 	@Test
-	public void testUpdateWhithNoExistingStudents(){
-		assertUpdate(false, 0);
+	public void testUpdateWhithNoExistingStudents() {
+		assertUpdate(false, 0, "0000", "matteo");
 	}
 
 	@Test
-	public void testUpdateWithExistingStudents(){
-		assertUpdate(true, 1);
+	public void testUpdateWithExistingStudents() {
+		assertUpdate(true, 1, "0000", "matteo");
 	}
-	
-	private void assertUpdate(boolean dbAnswer, int expectedVerify) {
-		when(myStudentsDB.exists("0000")).thenReturn(dbAnswer);
-		boolean result=studControl.update("0000","matteo");
+
+	private void assertUpdate(boolean dbAnswer, int expectedVerify, String id, String name) {
+		when(myStudentsDB.exists(id)).thenReturn(dbAnswer);
+		boolean result = studControl.update(id, name);
 		verify(myStudentsDB, times(expectedVerify)).updateDB(anyString(), anyString());
```

Comment:
see my previous comment, you should verify that the arguments passed to updateDB are exactly the ones you pass to update